test/e2e/framework/azure/eventgrid.go

```
@@ -0,0 +1,44 @@
+/*
+Copyright (c) 2021 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azure
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/eventgrid/mgmt/eventgrid"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/triggermesh/test-infra/test/e2e/framework"
+)
+
+func CreateEventGridComponents(ctx context.Context, subscriptionID, name, region, rg string) error {
+	egsubClient := eventgrid.NewEventSubscriptionsClient(subscriptionID)
+
+	authorizer, err := auth.NewAuthorizerFromEnvironment()
+	if err != nil {
+		framework.FailfWithOffset(3, "unable to create authorizer: %s", err)
+		return nil
+	}
+
+	egsubClient.Authorizer = authorizer
+
+	return nil
+}
+
+func DeleteEventGridComponents(ctx context.Context, eventGrid string) error {
+	return fmt.Errorf("unimplemented")
```

Similar comment as before. We typically _fail on error_ in helpers, because we want to avoid verbose error handling in the main test logic.
I know this one is special because it isn't used yet, but I don't think we'll want to return an error here.

Btw, if it's not used, do we need it?

---

You're absolutely right. I created the file thinking EventGrid was a dedicated component instead of integrated with the other parts of Azure. This has since been removed.